### ISSUE TITLE
Atom support for Mal is poor

### ISSUE THREAD

**Issue by wgmyers** [kanaka/mal#562]
I should know: such minimal support as exists - until now there wasn't any at all - is all my fault. Apologies in advance.

So fwiw I have implemented basic Atom support for Mal: https://atom.io/packages/language-lisp-mal

(Sadly, language-mal was taken, hence language-lisp-mal.)

In the unlikely event that the Venn diagram of people playing around with Mal and Atom users consists of anyone other than just me, this may be of interest, for certain values of 'interest'. Keywords, string constants and integers are supported. Reader macros not yet. User-defined anything, not really, except for the first instance of the thing defined by def! or defmacro!.

As an aside, implementing a syntax highlighting package for Mal was several orders of magnitude less fun than implementing Mal was.

Bug reports and pull requests welcome.

I'm about yay close to going back to Emacs at this point.

**Comment by sogaiu** [comment]
I don't know if you are interested in using tree-sitter, but FWIW, there is this: https://github.com/sogaiu/tree-sitter-mal

**Comment by wgmyers** [comment]
Thank you! That would certainly take the pain out of making a tree-sitter version: having to write a tree-sitter grammar first before going ahead with the syntax highlighter was why I chose to go the TextMate route.

**Comment by asarhaddon** [comment]
Please rename this issue.  There is an `atom` type in MAL with deliberately limited support, so the current title is confusing.

**Comment by wgmyers** [comment]
Apologies. Done.

**Comment by asarhaddon** [comment]
Apologies in return, then.  Your answer makes me realize that my tone was crude, please be indulgent to my english skills and have fun with MAL !

**Comment by wgmyers** [comment]
Not at all, no need to apologise. Thank you for pointing out the ambiguity - if I have been struggling to write anything useful in Mal, it is in part because I keep forgetting about Mal atoms...

**Comment by kanaka** [comment]
@wgmyers nice! Please send a PR with a modification to the README to add a link and brief description of this (probably to the "Other mal Projects" section).
